### PR TITLE
Make RegistrationAdministrationList sortable

### PR DESCRIPTION
**Pull request by FinnIckler**
Also automatically sorts by registered_on or paid_on, by default.

Sorting order actually doesn't change currently. I'll figure that out tomorrow

### CODE REVIEW

File: Frontend/src/pages/registration_administration/components/RegistrationAdministrationList.jsx [hunk 1]
```diff
@@ -44,6 +45,16 @@ const selectedReducer = (state, action) => {
   return newState
 }
 
+const sortReducer = createSortReducer([
```
Comment (kr-matthews) [inline Frontend/src/pages/registration_administration/components/RegistrationAdministrationList.jsx:48]: People like to sort by comment, for example for a multi-location FM competition where you need to put your location in the comment.
And what about DOB? I'm not sure how useful that is, but it's possible in the existing registration.

**Comment by FinnIckler** [comment]
I finally added some seed users and sorting works on all columns

File: Frontend/src/pages/registration_administration/components/RegistrationAdministrationList.jsx [hunk 2]
```diff
@@ -136,9 +159,47 @@ export default function RegistrationAdministrationList() {
...
+    return []
+  }, [registrationsWithUser, sortColumn, sortDirection])
+
   const { waiting, accepted, cancelled, pending } = useMemo(
-    () => partitionRegistrations(registrationsWithUser ?? []),
-    [registrationsWithUser],
+    () => partitionRegistrations(sortedRegistrationWithUser ?? []),
+    // this useMemo will not fire again if we only change the order of the array, which is why we need the sortColumn and sortDirection in the deps
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [sortedRegistrationWithUser, sortColumn, sortDirection],
```
Comment (gregorbg) [inline Frontend/src/pages/registration_administration/components/RegistrationAdministrationList.jsx:202]: I don't get this. `sortedRegistrationWithUser` is itself a `useMemo`, whose dependencies contain `sortColumn` and `sortDirection`.

You are arguing that this current memo needs to trigger if sort order changes, but when that happens the memo of `sortedRegistrationsWithUser` triggers, and by transitive rule this memo should then also trigger. So from what I can tell by staring at the code, your extra deps are totally unnecessary.

There has to be a cleaner solution than overriding good React principles and standards.

Comment (gregorbg) [reply Frontend/src/pages/registration_administration/components/RegistrationAdministrationList.jsx:202]: Wow, thanks for explaining, that would probably have taken me half an hour to figure out :smile: 
I'm tempted to start a rant why on earth JS sorts in-place, but I guess the short answer is "legacy reasons"...

Made a PR over at https://github.com/thewca/wca-registration/pull/490

Comment (kr-matthews) [reply Frontend/src/pages/registration_administration/components/RegistrationAdministrationList.jsx:202]: Half an hour wouldn't be so bad; it took me several months to understand why my code didn't work as expected when I first started React...

**Pull request merged**